test/functional/p2p_orphan_handling.py

```diff
@@ -789,6 +789,7 @@ def test_parents_change(self):
         # Disconnect peer1. peer2 should become the new candidate for orphan resolution.
         peer1.peer_disconnect()
         self.wait_until(lambda: node.num_test_p2p_connections() == 1)
+        peer2.sync_with_ping()  # Sync with the 'net' thread which completes the disconnection fully
         node.bumpmocktime(TXREQUEST_TIME_SKIP)
```

Comment:
it would be good to also add a comment in `disconnect_p2ps` that the wait there does not guarantee that the resources of all nodes (such as outstanding txrequests) are cleared.